src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,91 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    public static final int MIN_AGE = 18;
+    public static final int LOWER_THAN_RECOMMENDED_AGE = 17;
+    public static final String USER_LOGIN = "user";
+    public static final String VALID_PASSWORD = "validPassword";
+    private static RegistrationService registrationService;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setPassword(VALID_PASSWORD);
+        user.setLogin(USER_LOGIN);
+        user.setAge(MIN_AGE);
+    }
+
+    @Test
+    void register_validLogin_Ok() {
+        User actual = registrationService.register(user);
+        assertEquals(user, actual);
+    }
+
+    @Test
+    void register_invalidLogin_NotOk() {
+        user.setPassword("short");
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_invalidAge_NotOk() {
+        user.setAge(LOWER_THAN_RECOMMENDED_AGE);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_userNegativeAge_NotOk() {
+        user.setAge(-MIN_AGE);
```

Comment:
No need to check for negative age, it would be enough to check age just less than required